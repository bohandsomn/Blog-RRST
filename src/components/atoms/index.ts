export * from './Img'
export * from './P'
export * from './Icon'
export * from './Input'
export * from './Textarea'
export * from './Button'
export * from './Div'
export * from './Ul'
export * from './Li'
export * from './Label'
export * from './Form'