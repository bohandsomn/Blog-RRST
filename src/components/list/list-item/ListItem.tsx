import React, { useContext } from 'react'
import Toggle from '../../../context/toggle'
import useAppStyles from '../../../hooks/useAppStyles'
import { Div, Li, MainParagraph, SecondaryParagraph } from '../../atoms'

export const ListItem: React.FC<Props> = ({ children, right, left }) => {
    const {value, handleSetTrue, handleSetFalse} = useContext(Toggle)
    const className = useAppStyles(
        'border-radius--border-radius', 
        'color--main-color', 
        'display-flex', 
        'align-items-center', 
        'justify-content-space-between', 
        'padding-top-10', 
        'padding-right-10', 
        'padding-bottom-10', 
        'padding-left-10', 
        'cursor-pointer', 
        value ? 'background-color--main-bg-color' : undefined
    )
    const rightClassName = useAppStyles('margin-left-10')
    const leftClassName = useAppStyles('margin-right-10')

    return (
        <Li className={className} onMouseEnter={handleSetTrue} onMouseLeave={handleSetFalse}>
            {left && <Div className={leftClassName}>{left}</Div>}
            {value 
                ? <SecondaryParagraph>{children}</SecondaryParagraph>
                : <MainParagraph>{children}</MainParagraph>
            }
            {right && <Div className={rightClassName}>{right}</Div>}
        </Li>
    )
}

type Props = NonNullable<typeof Li.defaultProps> & {
    right?: JSX.Element
    left?: JSX.Element
}
