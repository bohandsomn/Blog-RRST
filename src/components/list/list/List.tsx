import React from 'react'
import useAppQuery from '../../../hooks/useAppQuery'
import { SecondaryBackground } from '../../../layouts/wrapper/background'
import { Ul } from '../../atoms'

export const List: React.FC<Props> = ({ children }) => {
    const query = useAppQuery('list')
    return (
        <SecondaryBackground>
            <Ul query={query}>
                {children}
            </Ul>
        </SecondaryBackground>
    )
}

type Props = NonNullable<typeof Ul.defaultProps>
