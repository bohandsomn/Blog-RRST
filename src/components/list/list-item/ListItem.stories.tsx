import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { ListItem } from './ListItem'
import ToggleProvider from '../../../provider/toggle'
import { Account } from '../../atoms'
import BodyStorybook from '../../../service/component-explorer/BodyStorybook'
import { ComponentPropsStorybook } from '../../../utility/type'
import AtomProviderStorybook from '../../../service/component-explorer/AtomProviderStorybook'

export default {
  title: 'Molecule/List/ListItem',
  component: ListItem,
} as ComponentMeta<typeof ListItem>

const Template: ComponentStory<ComponentPropsStorybook<typeof ListItem.defaultProps>> = ({theme, ...args}) => {
    return (
        <BodyStorybook theme={theme}>
            <AtomProviderStorybook>
                <ToggleProvider>
                    <ListItem {...args} />
                </ToggleProvider>
            </AtomProviderStorybook>
        </BodyStorybook>
    )
}

export const RightDark = Template.bind({})
RightDark.args = {
    children: 'RightDark',
    theme: 'dark',
    right: <Account />
}

export const LeftDark = Template.bind({})
LeftDark.args = {
    children: 'LeftDark',
    theme: 'dark',
    left: <Account />
}

export const RightLight = Template.bind({})
RightLight.args = {
    children: 'RightLight',
    theme: 'light',
    right: <Account />
}

export const LeftLight = Template.bind({})
LeftLight.args = {
    children: 'LeftLight',
    theme: 'light',
    left: <Account />
}