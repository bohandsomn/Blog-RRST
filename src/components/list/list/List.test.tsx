import { fireEvent, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { List } from './List'
import { Account } from '../../atoms'
import ToggleProvider from '../../../provider/toggle'
import { ListItem } from '../list-item'

describe('ListItem', () => {
    test('should renders the element', () => {
        render(
            <List>
                <ToggleProvider>
                    <ListItem right={<Account />}>Account</ListItem>
                </ToggleProvider>
            </List>
        )
        const element = screen.getByText(/Account/)
        expect(element).toBeInTheDocument()
    })

    test('should containg svg', () => {
        render(
            <List>
                <ToggleProvider>
                    <ListItem right={<Account data-testid="svg" />}>Account</ListItem>
                </ToggleProvider>
            </List>
        )
        const element = screen.getByTestId('svg')
        expect(element).toBeInTheDocument()
    })
    
    test('shouldn\'t changes snapshots', () => {
        render(
            <List>
                <ToggleProvider>
                    <ListItem right={<Account />}>Account</ListItem>
                </ToggleProvider>
            </List>
        )
        const element = screen.getByRole('list')
        expect(element).toMatchSnapshot()
    })
})