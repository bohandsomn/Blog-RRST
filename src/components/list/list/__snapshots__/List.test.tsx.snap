// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem shouldn't changes snapshots 1`] = `
<ul
  class="sc-ftTHYK"
>
  <li
    class="sc-pyfCe"
  >
    <p
      class="sc-gswNZR"
    >
      Account
    </p>
    <div
      class="sc-iBYQkv"
    >
      <svg
        class="sc-hLBbgP"
        fill="none"
        height="32"
        stroke="var(--main-color)"
        viewBox="0 0 32 32"
        width="32"
      >
        <path
          class="sc-dkrFOg"
          d="M26.6666 28V25.3333C26.6666 23.9188 26.1047 22.5623 25.1045 21.5621C24.1043 20.5619 22.7477 20 21.3333 20H10.6666C9.2521 20 7.89554 20.5619 6.89535 21.5621C5.89515 22.5623 5.33325 23.9188 5.33325 25.3333V28"
        />
        <path
          class="sc-dkrFOg"
          d="M16.0001 4C14.5856 4 13.229 4.5619 12.2288 5.5621C11.2287 6.56229 10.6667 7.91885 10.6667 9.33333C10.6667 10.7478 11.2287 12.1044 12.2288 13.1046C13.229 14.1048 14.5856 14.6667 16.0001 14.6667C17.4146 14.6667 18.7711 14.1048 19.7713 13.1046C20.7715 12.1044 21.3334 10.7478 21.3334 9.33333C21.3334 7.91885 20.7715 6.56229 19.7713 5.5621C18.7711 4.5619 17.4146 4 16.0001 4Z"
        />
      </svg>
    </div>
  </li>
</ul>
`;
